src/test/java/com/snowflake/kafka/connector/ConnectClusterBaseIT.java

```diff
@@ -0,0 +1,113 @@
+package com.snowflake.kafka.connector;
+
+
+import com.snowflake.kafka.connector.fake.SnowflakeFakeSinkConnector;
+import com.snowflake.kafka.connector.fake.SnowflakeFakeSinkTask;
+import org.apache.kafka.connect.runtime.AbstractStatus;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorStateInfo;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.storage.StringConverter;
+import org.apache.kafka.connect.util.clusters.EmbeddedConnectCluster;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.BUFFER_COUNT_RECORDS;
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC;
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.NAME;
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.SNOWFLAKE_DATABASE;
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.SNOWFLAKE_PRIVATE_KEY;
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.SNOWFLAKE_SCHEMA;
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.SNOWFLAKE_URL;
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.SNOWFLAKE_USER;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.CONNECTOR_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.KEY_CONVERTER_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.TASKS_MAX_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.VALUE_CONVERTER_CLASS_CONFIG;
+import static org.apache.kafka.connect.sink.SinkConnector.TOPICS_CONFIG;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+public class ConnectClusterBaseIT {
+
+    protected static EmbeddedConnectCluster connectCluster;
+
+    protected static final String TEST_TOPIC = "kafka-int-test";
+    protected static final String TEST_CONNECTOR_NAME = "test-connector";
+    protected static final Integer TASK_NUMBER = 1;
+    private static final Duration CONNECTOR_MAX_STARTUP_TIME = Duration.ofSeconds(20);
+
+    @Before
+    public void setup() {
+        if (connectCluster == null) {
+            connectCluster = new EmbeddedConnectCluster.Builder().name("kafka-push-connector-connect-cluster").numWorkers(3).build();
+            connectCluster.start();
+            connectCluster.kafka().createTopic(TEST_TOPIC);
+            connectCluster.configureConnector(TEST_CONNECTOR_NAME, createProperties());
+            await().timeout(CONNECTOR_MAX_STARTUP_TIME).until(this::isConnectorRunning);
+        }
```

Comment:
Nit: will it be better to move the connector setup into @BeforeClass? Looks like we can reuse the same connector between tests.

Reply:
yea, i'm not sure there, it's been a while since i used junit < 5. The problem is with createProperties as it's not static a i thought maybe it's gonna be overridden later but i guess I'm thinking too much here and  this can be refactor later